genesyscloud/task_management_workbin/resource_genesyscloud_task_management_workbin.go

```diff
@@ -0,0 +1,137 @@
+*/
+
+// getAllAuthTaskManagementWorkbin retrieves all of the task management workbin via Terraform in the Genesys Cloud and is used for the exporter
+func getAllAuthTaskManagementWorkbins(ctx context.Context, clientConfig *platformclientv2.Configuration) (resourceExporter.ResourceIDMetaMap, diag.Diagnostics) {
+	proxy := getTaskManagementWorkbinProxy(clientConfig)
+	resources := make(resourceExporter.ResourceIDMetaMap)
+
+	workbins, err := proxy.getAllTaskManagementWorkbin(ctx)
+	if err != nil {
+		return nil, diag.Errorf("failed to get all workbins: %v", err)
+	}
+
+	for _, workbin := range *workbins {
+		log.Printf("Dealing with task management workbin id: %s", *workbin.Id)
+		resources[*workbin.Id] = &resourceExporter.ResourceMeta{Name: *workbin.Id}
+	}
+
+	return resources, nil
+}
+
+// createTaskManagementWorkbin is used by the task_management_workbin resource to create Genesys cloud task management workbin
+func createTaskManagementWorkbin(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*gcloud.ProviderMeta).ClientConfig
+	proxy := getTaskManagementWorkbinProxy(sdkConfig)
+
+	taskManagementWorkbin := platformclientv2.Workbincreate{
+		Name:        platformclientv2.String(d.Get("name").(string)),
+		DivisionId:  platformclientv2.String(d.Get("division_id").(string)),
+		Description: platformclientv2.String(d.Get("description").(string)),
+	}
+
+	log.Printf("Creating task management workbin %s", *taskManagementWorkbin.Name)
+	workbin, err := proxy.createTaskManagementWorkbin(ctx, &taskManagementWorkbin)
+	if err != nil {
+		return diag.Errorf("failed to create task management workbin: %s", err)
+	}
+
+	d.SetId(*workbin.Id)
+	log.Printf("Created task management workbin %s", *workbin.Id)
+	return readTaskManagementWorkbin(ctx, d, meta)
+}
+
+// readTaskManagementWorkbin is used by the task_management_workbin resource to read an task management workbin from genesys cloud
+func readTaskManagementWorkbin(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*gcloud.ProviderMeta).ClientConfig
+	proxy := getTaskManagementWorkbinProxy(sdkConfig)
+
+	log.Printf("Reading task management workbin %s", d.Id())
+
+	return gcloud.WithRetriesForRead(ctx, d, func() *retry.RetryError {
+		workbin, respCode, getErr := proxy.getTaskManagementWorkbinById(ctx, d.Id())
+		if getErr != nil {
+			if gcloud.IsStatus404ByInt(respCode) {
+				return retry.RetryableError(fmt.Errorf("failed to read task management workbin %s: %s", d.Id(), getErr))
+			}
+			return retry.NonRetryableError(fmt.Errorf("failed to read task management workbin %s: %s", d.Id(), getErr))
+		}
+
+		cc := consistency_checker.NewConsistencyCheck(ctx, d, meta, ResourceTaskManagementWorkbin())
+
+		resourcedata.SetNillableValue(d, "name", workbin.Name)
+		resourcedata.SetNillableValue(d, "division_id", workbin.Division.Id)
```

Comment:
I think this might throw an error if `Division` is nil. There should be similar function for handling reference objects

Reply:
Add a `resourcedata` function: `SetNillableReferenceDivision` because existing `SetNillableReference` and `SetNillableReferenceWritableDivision` does not apply